petab/parameter_mapping.py

```diff
@@ -281,49 +298,54 @@ def _apply_dynamic_parameter_overrides(mapping: ParMappingDict,
     for overridee_id in condition_df.columns:
         if overridee_id == CONDITION_NAME:
             continue
-        if condition_df[overridee_id].dtype != 'O':
-            continue
 
-        overrider_id = condition_df.loc[condition_id, overridee_id]
-        mapping[overridee_id] = overrider_id
+        mapping[overridee_id] = condition_df.loc[condition_id, overridee_id]
+
 
+def _apply_parameter_table(mapping: ParMappingDict,
+                           parameter_df: Optional[pd.DataFrame] = None
+                           ) -> None:
+    """Replace parameters from parameter table in mapping list for a given
+    condition.
 
-def fill_in_nominal_values(mapping: ParMappingDict,
-                           parameter_df: pd.DataFrame) -> None:
-    """Replace non-estimated parameters in mapping list for a given condition
-    by ``nominalValues`` provided in parameter table.
+    Replace non-estimated parameters by ``nominalValues``
+    (un-scaled / lin-scaled), replace estimated parameters by the respective
+    ID.
 
     Arguments:
         mapping:
-            mapping dict obtained from get_parameter_mapping_for_condition
+            mapping dict obtained from ``get_parameter_mapping_for_condition``
         parameter_df:
             PEtab parameter table
     """
 
     if parameter_df is None:
         return
-    if ESTIMATE not in parameter_df:
-        return
-
-    overrides = {row.name: row.nominalValue for _, row
-                 in parameter_df.iterrows() if row.estimate != 1}
 
-    for par, overridee in mapping.items():
-        if not isinstance(overridee, str):
+    for row in parameter_df.itertuples():
+        if row.Index not in mapping:
+            # The current parameter is not required for this condition
             continue
+        if ESTIMATE in row._fields and getattr(row, ESTIMATE) == 0:
+            mapping[row.Index] = getattr(row, NOMINAL_VALUE)
+        else:
+            mapping[row.Index] = row.Index
```

Comment:
not completely sure, but looks reasonable ...

Reply:
If the parameter is not to be estimated and there is a nominal, we use that. Otherwise we leave the parameter name, indicating that it is to be estimated.